ADBench/run-all.ps1

```diff
@@ -1,29 +1,109 @@
-function assert($expr)
-{
-    if (!(& $expr)) {
+# Accept command-line args
+param([int]$global:nruns_f=10, [int]$global:nruns_J=10, [string]$global:tmpdir="")
+
+# Assert function
+function assert ($expr) {
+    if (!(& $expr @args)) {
         throw "Assertion failed [$expr]"
     }
 }
 
-$dir = split-path ($MyInvocation.MyCommand.Path)
-assert { $dir -match 'ADbench$' }
-$dir = Split-Path $dir
+# Get source dir
+
+$global:dir = split-path ($MyInvocation.MyCommand.Path)
+assert { $global:dir -match 'ADbench$' }
+$global:dir = Split-Path $global:dir
+
+Write-Host "$global:dir"
+
+# Load bindir variable
+assert "Test-Path" "$global:dir/ADBench/cmake_vars.ps1"
+. $global:dir/ADBench/cmake_vars.ps1
+
+# Set tmpdir default
+if (!$global:tmpdir) { $global:tmpdir = "$dir/tmp" }
+
+
+# Custom Tool class
+Class Tool {
+	[string]$name
+	[bool]$gmm_both
+	[string]$type
+	[bool]$gmm_use_defs
+
+	# Static constants
 
-Write-Host "$dir"
+	# GMM values - NOTE not all implemented for ceres
+	static [int[]]$gmm_d_vals = @(2, 10) # @(2, 10, 20, 32, 64)
+	static [int[]]$gmm_k_vals = @(5, 10, 25) # @(5, 10, 25, 50, 100, 200)
 
-# $bindir = 'C:\Users\Andrew Fitzgibbon\CMakeBuilds\9f453784-edfa-5e36-80b5-7cbdb5076dfb\build\x86-Debug'
-$bindir = 'C:\Users\Zak Smith\CMakeBuilds\95e43dd6-1979-0633-8dca-9ab4e04499c8\build\x64-Debug'
+	static [string]$gmm_dir_in = "$dir/data/gmm/1k/"
 
-function run($tool, $objective)
-{
-    & "$bindir\tools\$tool\Tools-$tool-$objective.exe" @args
+	# Constructor
+	Tool ([string]$name, [bool]$gmm_both, [string]$type, [bool]$gmm_defs) {
+		$this.name = $name
+		$this.gmm_both = $gmm_both
+		$this.type = $type
+		$this.gmm_use_defs = $gmm_defs
+	}
+
+	# Run all tests for this tool
+	[void] runall () {
+		Write-Host $this.name
+		$this.testgmm("")
+	}
+
+	# Run a single test
+	[void] run ([string]$objective, [string]$dir_in, [string]$dir_out, [string]$fn) {
+		$cmd = ""
+		$cmdargs = @($dir_in, $dir_out, $fn, $global:nruns_f, $global:nruns_J)
```

Comment:
Ah OK, I see why you wanted it global.  You should be able to do $script:nruns_f here.